src/pal/test/snp_ioctl_test.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+
+#include "ccf/crypto/symmetric_key.h"
+#include "ccf/ds/logger.h"
+#include "ccf/pal/snp_ioctl.h"
+#include "crypto/openssl/hash.h"
+
+#include <random>
+#include <span>
+#include <string>
+
+#define DOCTEST_CONFIG_IMPLEMENT
+#include <doctest/doctest.h>
+
+TEST_CASE("SNP derive key")
+{
+  using namespace ccf::pal;
+  auto key1 = snp::make_derived_key();
+  auto key2 = snp::make_derived_key();
```

Comment:
```suggestion
  auto key2 = snp::make_derived_key();
  
  REQUIRE_EQ(ccf::ds::to_hex(key1->get_raw()), ccf::ds::to_hex(key2->get_raw()));
```

I think it's useful to clarify in this test whether multiple calls (with the same parameters) get back _exactly the same value_ or _keys that can mutually verify each other_. I'm confident it's the former, because we're using it for encryption and decryption, but we should confirm that here.

Reply:
Done!